kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -258,28 +261,43 @@ private Optional<OwnerReference> extractOwnerRefFromKubernetesService(Service se
-        if (clusterResolutionResult.anyDependenciesNotFoundFor(clusterRef)) {
+        var unresolvedIngressProxies = clusterResolutionResult.allDanglingReferences()
+                .filter(DanglingReference.hasReferrerKind(KAFKA_PROXY_INGRESS_KIND).and(DanglingReference.hasReferentKind(KAFKA_PROXY_KIND))).collect(Collectors.toSet());
+        if (clusterResolutionResult.allDanglingReferences().anyMatch(DanglingReference.hasReferrer(clusterRef))) {
             Stream<String> proxyMsg = refsMessage("spec.proxyRef references ", cluster,
-                    clusterResolutionResult.findDanglingReferences(clusterRef, KAFKA_PROXY_KIND));
+                    clusterResolutionResult.allDanglingReferences()
+                            .filter(DanglingReference.hasReferrer(clusterRef).and(DanglingReference.hasReferentKind(KAFKA_PROXY_KIND)))
+                            .map(DanglingReference::absentRef));
             Stream<String> serviceMsg = refsMessage("spec.targetKafkaServiceRef references ", cluster,
-                    clusterResolutionResult.findDanglingReferences(clusterRef, KAFKA_SERVICE_KIND));
+                    clusterResolutionResult.allDanglingReferences()
+                            .filter(DanglingReference.hasReferrer(clusterRef).and(DanglingReference.hasReferentKind(KAFKA_SERVICE_KIND)))
+                            .map(DanglingReference::absentRef));
             Stream<String> ingressMsg = refsMessage("spec.ingresses[].ingressRef references ", cluster,
-                    clusterResolutionResult.findDanglingReferences(clusterRef, KAFKA_PROXY_INGRESS_KIND));
+                    clusterResolutionResult.allDanglingReferences()
+                            .filter(DanglingReference.hasReferrer(clusterRef).and(DanglingReference.hasReferentKind(KAFKA_PROXY_INGRESS_KIND)))
+                            .map(DanglingReference::absentRef));
             Stream<String> filterMsg = refsMessage("spec.filterRefs references ", cluster,
-                    clusterResolutionResult.findDanglingReferences(clusterRef, KAFKA_PROTOCOL_FILTER_KIND));
+                    clusterResolutionResult.allDanglingReferences()
+                            .filter(DanglingReference.hasReferrer(clusterRef).and(DanglingReference.hasReferentKind(KAFKA_PROTOCOL_FILTER_KIND)))
+                            .map(DanglingReference::absentRef));
             return statusFactory.newFalseConditionStatusPatch(cluster, Condition.Type.ResolvedRefs, Condition.REASON_REFS_NOT_FOUND,
                     joiningMessages(proxyMsg, serviceMsg, ingressMsg, filterMsg));
         }
-        else if (clusterResolutionResult.anyResolvedRefsConditionsFalse() || !unresolvedIngressProxies.isEmpty()) {
+        else if (clusterResolutionResult.allResolvedReferents().anyMatch(ResourcesUtil::hasResolvedRefsFalseCondition) || !unresolvedIngressProxies.isEmpty()) {
```

Comment:
Again I'd like a method to wrap this compound condition to give it an intention revealing name.